testing/trino-testing/src/main/java/io/trino/testing/BaseTestParquetWithBloomFilters.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing;
+
+import io.trino.Session;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class BaseTestParquetWithBloomFilters
+        extends AbstractTestQueryFramework
+{
+    protected Path dataDirectory;
+    private static final String COLUMN_NAME = "dataColumn";
+    // containing extreme values, so the row group cannot be eliminated by the column chunk's min/max statistics
+    private static final List<Integer> TEST_VALUES = Arrays.asList(Integer.MIN_VALUE, Integer.MAX_VALUE, 1, 3, 7, 10, 15);
+    private static final int MISSING_VALUE = 0;
+
+    @Test
+    public void verifyBloomFilterEnabled()
+    {
+        assertThat(query(format("SHOW SESSION LIKE '%s.parquet_use_bloom_filter'", getSession().getCatalog().orElseThrow())))
+                .skippingTypesCheck()
+                .matches(result -> result.getRowCount() == 1)
+                .matches(result -> {
+                    String value = (String) result.getMaterializedRows().get(0).getField(1);
+                    return value.equals("true");
+                });
+    }
+
+    @Test
+    public void testBloomFilterRowGroupPruning()
+    {
+        String tableName = createParquetTableWithBloomFilter(COLUMN_NAME, TEST_VALUES);
+
+        // assert table get populated with data
+        assertQueryStats(
+                getSession(),
+                "SELECT * FROM " + tableName,
+                queryStats -> {},
+                results -> assertThat(results.getRowCount()).isEqualTo(TEST_VALUES.size()));
```

Comment:
you can assert on the result of a COUNT(*) query instead

Reply:
done.